### ISSUE TITLE
Bug: [consistent-type-exports] Importing a shadowing type breaks consistent-type-imports/exports

### ISSUE THREAD

**Issue by simon-abbott** [typescript-eslint/typescript-eslint#11761]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

The following piece of code has two issues:

```ts
import { SomeThing } from './types.js';

const SomeThing = { foo: 'bar' };

export { SomeThing };
```

Firstly, it does not flag that `import { SomeThing } from './types.js';` should be `import { type SomeThing } from './types.js';` (problem with `consistent-type-imports`).

Secondly, it does error with "All exports in the declaration are only used as types.", even though the export is very clearly _not_ a type.

This issue manifested itself in v8.43.0. I suspect it is a regression to do with https://github.com/typescript-eslint/typescript-eslint/pull/11457

### Reproduction Repository Link

https://github.com/simon-abbott/ts-eslint-issue-mre-import-shadowing

### Repro Steps

1) clone the repo
2) `npm install`
3) `npm run lint`

Running `npm run lint` _should_ error on the `import` in `src/index.ts`, and _should not_ error on the export. Instead it does the reverse.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.46.4` |
| `@typescript-eslint/parser`            | `8.46.4` |
| `TypeScript`                           | `5.9.2` |
| `ESLint`                               | `8.57.1` |
| `node`                                 | `22.19.0` |

**Comment by bradzacher** [comment]
The import rule not firing is expected - based on pure syntactic analysis the imported name could be anything.

Purely based on usage - the only way that the name is used is as a value in the export statement.

You're suggesting that the rule should understand that you have a type that is shadowing the local value - but that's not a safe assumption to make. The name could be an imported type - or it could be an imported value that will cause a TS error.

Because the usage is ambiguous - we do not report on it.

**Comment by bradzacher** [comment]
The exports rule is a regression though

**Comment by simon-abbott** [comment]
> The import rule not firing is expected - based on pure syntactic analysis the imported name could be anything.
> 
> Purely based on usage - the only way that the name is used is as a value in the export statement.
> 
> You're suggesting that the rule should understand that you have a type that is shadowing the local value - but that's not a safe assumption to make. The name could be an imported type - or it could be an imported value that will cause a TS error.
> 
> Because the usage is ambiguous - we do not report on it.

Ah, fair enough. I had expected that since it was exported using `export type` that the system could tell it was not a value, but I see now that I was misremembering how that rule worked.

**Comment by ntdiary** [comment]
> The exports rule is a regression though

Hm, yeah, I'm running some tests now, will raise a PR later. :)

**Comment by ntdiary** [comment]
I placed the check for `SymbolFlags.Value` after the iteration, rather than during it, which caused this regression. 😅
